test-data/unit/check-statements.test

```diff
@@ -1514,3 +1514,10 @@ class A(): pass
 class B(): pass
 [out]
 main:8: error: Incompatible types in assignment (expression has type "A", variable has type "B")
+
+[case testAugmentedAssignmentIntFloat]
+weight0 = 65.5
```

Comment:
After I did it, I discovered another problem: I don't update the binder (the last `weight0` should go back to `float`, but it stayed as `int` in my patch). I fixed it by using a higher level `check_assignment` method instead of copying/pasting a line from `check_simple_assignment` . I think this way, the code became cleaner overall since I could get rid of a couple lines from the original implementation of `visit_operator_assignment_stmt` (they are taken care of in the `check_assignment` call).